plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -412,18 +418,23 @@ public List<TableInfo> getTables(String databaseName)
     private List<TableInfo> getTablesInternal(Consumer<Table> cacheTable, String databaseName)
     {
         try {
-            return stats.getGetTables()
+            ImmutableList<software.amazon.awssdk.services.glue.model.Table> glueTables = stats.getGetTables()
                     .call(() -> glueClient.getTablesPaginator(builder -> builder
                                     .applyMutation(glueContext::configureClient)
                                     .databaseName(databaseName)).stream()
                             .map(GetTablesResponse::tableList)
                             .flatMap(List::stream))
-                    .filter(tableVisibilityFilter)
-                    .map(glueTable -> GlueConverter.fromGlueTable(glueTable, databaseName))
-                    .peek(cacheTable)
+                    .filter(tableVisibilityFilter).collect(toImmutableList());
+
+            // Store in cache only valid tables
+            glueTables.stream()
+                    .flatMap(glueTable -> convertFromGlueIgnoringErrors(glueTable, databaseName))
```

Comment:
seems like an overkill:

return Optional<Table> from convertFromGlueIgnoringErrors

and then:
```
// Store in cache only valid tables
            for (software.amazon.awssdk.services.glue.model.Table table : glueTables) 
            {
                convertFromGlueIgnoringErrors(table, databaseName)
                        .ifPresent(cacheTable);
            }
```